src/Module/Special/HTTPException.php

```diff
@@ -33,27 +36,52 @@
  */
 class HTTPException
 {
+	/** @var L10n */
+	protected $l10n;
+	/** @var LoggerInterface */
+	protected $logger;
+	/** @var Arguments */
+	protected $args;
+	/** @var bool */
+	protected $isSiteAdmin;
+	/** @var array */
+	protected $server;
+	/** @var string */
+	protected $requestId;
+
+	public function __construct(L10n $l10n, LoggerInterface $logger, Arguments $args, UserSession $session, Request $request, array $server = [])
```

Comment:
Shouldn't this class inherit from `BaseModule`? 🤔

Reply:
No, because it must not be used at the normal `BaseModule::run()` and not retrieved by the `App\Router`. It's directly called with an exception.

And additionally, if I do inherit it from `BaseModule`, I cannot use the `rawContent()` and `content()` calls anymore, because they're called with an `HTTPException` instead of `array $request`.

So it make things more complicated.

Another proof for me that there's a design flaw at `BaseModule`...